src/ol/geom/flat/CircularArc.js

```diff
@@ -0,0 +1,266 @@
+
+  /**
+   * Computes and returns the point's magnitude.
+   * @return {number} The computed magnitude.
+   */
+  magnitude = () => Math.sqrt(this.x * this.x + this.y * this.y);
+
+  /**
+   * Rotates the point 90 degrees in clockwise direction and returns the
+   * result.
+   * @return {Point} The resulting point.
+   */
+  rotated90ClockWise = () => new Point(-this.y, this.x);
+
+  /**
+   * Tests if this point and the given other point are equal in terms of
+   * coordinates. Note that they are required to be exactly equal in order
+   * for them to be considered equal.
+   * @param {Point} other The given other point.
+   * @return {boolean} True if equal, false otherwise.
+   */
+  equals = (other) => this.x === other.x && this.y === other.y;
+
+  /**
+   * Computes and returns the normalized version of this point.
+   * @return {Point} The normalized point.
+   */
+  normalized = () => {
+    const magnitude = this.magnitude();
+    return new Point(this.x / magnitude, this.y / magnitude);
+  };
+}
+
+export class Line {
+  /**
+   * Constructs a new Line given a begin and end point.
+   * @param {Point} begin The given begin point.
+   * @param {Point} end The given end point.
+   */
+  constructor(begin, end) {
+    this.begin = begin;
+    this.end = end;
+  }
+
+  /**
+   * Computes and returns the center point of the line.
+   * @return {Point} The center point.
+   */
+  center = () => {
+    return this.begin.add(
+      this.end
+        .subtract(this.begin)
+        .normalized()
+        .times(0.5 * this.length())
+    );
+  };
+
+  /**
+   * Computes and returns the length of the line.
+   * @return {number} The length.
+   */
+  length = () => this.end.subtract(this.begin).magnitude();
+
+  /**
+   * Computes and returns a unit vector (point) in direction of the line's end
+   * point.
+   * @return {Point} The computed unit vector (point).
+   */
+  unit = () => this.end.subtract(this.begin).normalized();
+
+  /**
+   * Computes and returns the point at which this line and a given other line
+   * would intersect, if any. If no such intersection can be computed an
+   * exception is thrown. The returned point is the point where they would
+   * intersect if they don't actually intersect as of right now.
+   * @param {Line} other The given other line.
+   * @return {Point} The point of intersection.
+   */
+  intersection = (other) => {
+    // source:
+    // https://dirask.com/posts/JavaScript-how-to-calculate-intersection-point-of-two-lines-for-given-4-points-VjvnAj
+
+    const p1 = this.begin;
+    const p2 = this.end;
+    const p3 = other.begin;
+    const p4 = other.end;
+
+    // down part of intersection point formula
+    const d1 = (p1.x - p2.x) * (p3.y - p4.y); // (x1 - x2) * (y3 - y4)
+    const d2 = (p1.y - p2.y) * (p3.x - p4.x); // (y1 - y2) * (x3 - x4)
+    const d = d1 - d2;
+
+    if (d === 0) {
+      throw new Error('Number of intersection points is zero or infinity.');
```

Comment:
Hier kom je ook terecht bij collineaire bogen (rechte lijn). Als we dat voor de aanroep van deze functie kunnen detecteren dan kan de 'boog' als 2 rechte lijnstukken getekend worden.

Reply:
Zouden we dat het beste bij het inlezen al oplossen denk je? Dus in `WKB`.